scanners/ssh-audit/parser/parser.js

```diff
@@ -0,0 +1,262 @@
+    Object.entries(value).map(([recommendationAction, algorithms]) => {
+        //algorithmType = kex/ key/ mac, , algorithmNames = {name+note}
+        Object.entries(algorithms).map(([algorithmType, algorithmData]) => {
+            const algorithmNames = []
+            Object.entries(algorithmData).flatMap(([keyNames, content]) => { algorithmNames.push(Object.values(content)) })
+            var action = "";
+            if (recommendationAction == "del" && recommendationSeverityLevel == "critical") action = "delCritical"
+            else if (recommendationAction == "del" && recommendationSeverityLevel == "warning") action = "delWarning"
+            else if (recommendationAction == "chg" && recommendationSeverityLevel == "critical") action = "chgCritical"
+            else if (recommendationAction == "chg" && recommendationSeverityLevel == "warning") action = "chgWarning"
+            else if (recommendationAction == "add" && recommendationSeverityLevel == "critical") action = "addCritical"
+            else if (recommendationAction == "add" && recommendationSeverityLevel == "warning") action = "addWarning"
+            const findingTemplate = templates[action][algorithmType] || null;
+
+            if (findingTemplate != null && typeof (findingTemplate) != "undefined") {
+                findingTemplate['severity'] = severity
+                findingTemplate['category'] = "SSH Policy Violation"
+                
+                combinedAlgorithmNames = []
+                algorithmNames.map(([algName, note]) => {
+                    if (note == "") combinedAlgorithmNames.push(algName)
+                    else combinedAlgorithmNames.push((algName + " (Note: " + note + ")"))
+                })
+
+                findingTemplate['algorithms'] = combinedAlgorithmNames.flat()
+                policyViolationFindings.push(findingTemplate)
+            }
+        })
+    })
+
+    return policyViolationFindings;
+}
+
+/**
+ * Transforms cves's from the ssh-audit scanner into SSH Violation Findings
+ * @param {{}} cves 
+ */
+function transformCVEtoFinding(cves) {
+
+    const cvesArray = Object.values(cves)
+    const cvesFindings = []
+    var severity = ''
+    Object.values(cvesArray).flatMap(({cvssv2, description, name}) => {
+        const findingTemplate = {}
+        if (cvssv2 < 4) severity = "LOW"
+        else if (cvssv2 < 7) severity = "MEDIUM"
+        else severity = "HIGH"
+        findingTemplate['name'] = name
+        findingTemplate['description'] = description
+        findingTemplate['category'] = "SSH Violation"
+        findingTemplate['severity'] = severity
+        findingTemplate['cvssv2'] = cvssv2
+
+        findingTemplate['references'] = []
+        findingTemplate['references'].push({'type':'CVE', 'value':`${name}`})
+        findingTemplate['references'].push({'type':'URL', 'value':`https://nvd.nist.gov/vuln/detail/${name}`})
+        cvesFindings.push(findingTemplate)
+    })
+    return cvesFindings;
+}
```

Comment:
Can also be simplified by directly using the map result and simplying the finding template by just returning it directly instead of gradually building it up.

```suggestion
function transformCVEtoFinding(cves) {
    const cvesArray = Object.values(cves)
    return Object.values(cvesArray).flatMap(({cvssv2, description, name}) => {
        let severity = 'HIGH'
        if (cvssv2 < 4) severity = "LOW"
        else if (cvssv2 < 7) severity = "MEDIUM"

        return {
            name,
            description,
            category: "SSH Violation",
            severity,
            cvssv2,
            references: [
                {type: 'CVE', value: `${name}`},
                {type: 'URL', value: `https://nvd.nist.gov/vuln/detail/${name}`}
            ],
        };
    });
}
```

Reply:
At the moment this calls Object.values twice. I wasn't sure if that was intentional or a misstake

Reply:
Thank you, it calls Object.values twice because of the way the object is built. We have to extract the needed information twice.